tola/tests/test_oauth.py

```diff
@@ -14,25 +16,28 @@ class OAuthTest(TestCase):
     Test cases for OAuth Provider interface
     """
 
-    fixtures = ['fixtures/users.json','fixtures/oauth_test_data.json']
+    def setUp(self):
+        self.tola_user = factories.TolaUser()
+        self.app = factories.Application(user=self.tola_user.user)
+        self.grant = factories.Grant(application=self.app,
+                                     user=self.tola_user.user)
 
     def test_authorization(self):
         """
         Tests if the simple search responds
-        :return: 
+        :return:
         """
-        c = Client(HTTP_USER_AGENT='Test/1.0')
-        c.login(username='test', password='1234')
+        self.tola_user.user.set_password('1234')
+        self.tola_user.user.save()
 
-        oauth_app = Application.objects.all()[0]
+        c = Client(HTTP_USER_AGENT='Test/1.0')
+        c.login(username=self.tola_user.user.username, password='1234')
 
         # Get Authorization token
         authorize_url = '/oauth/authorize?state=random_state_string' \
-                        '&client_id='+oauth_app.client_id+'' \
-                        '&response_type=code'
+                        '&client_id=%s&response_type=code' % self.app.client_id
```

Comment:
It's in general better to use `format()`, or at least we should be moving forward that and using it everywhere. Less error-prone, more efficient, powerful and readable in some cases.

[PEP 3101](https://www.python.org/dev/peps/pep-3101/) proposes the replacement of the % operator with the new, advanced string formatting in Python 3, where it would be the default.

Reply:
I don't think it's a problem in a test but I'm gonna change it.

Reply:
Yeah, in a test it doesn't matter. But it's easier to do it everywhere the same than having it mixed